go.mod

```diff
@@ -7,7 +7,7 @@ require (
 	github.com/filecoin-project/go-address v0.0.5
 	github.com/filecoin-project/go-dagaggregator-unixfs v0.2.0
 	github.com/filecoin-project/go-indexer-core v0.2.15
-	github.com/filecoin-project/go-legs v0.3.14
+	github.com/filecoin-project/go-legs v0.3.15-0.20220523114716-031a27fc689d
```

Comment:
This is a [commit from `main`](https://github.com/filecoin-project/go-legs/commit/031a27fc689d1e1339638ad225118f62efd3bacd) branch in`go-legs`. The concrete version is pending the review and merge of https://github.com/filecoin-project/go-legs/pull/127.

In the meantime, I am going ahead with the merge of this PR to get the memory fix deployed on `dev` cluster and start the full sync from NFT storage provider.

Cc @willscott @gammazero